99_helpers/create_sample_data.R

```diff
@@ -0,0 +1,355 @@
+        sector
+      )
+
+    df_data <- temp %>%
+      filter(adm0_pcode == sample(unique(temp$adm0_pcode), 1)) %>%
+      pivot_wider(
+        names_from = sector,
+        values_from = pin,
+        values_fill = 0
+      )
+
+    if (gender_age_disagg & pop_group_disagg) { # nolint
+      df_data <- df_data %>%
+        separate(disaggregation,
+          into = c(
+            "Population group",
+            "Sex",
+            "Age"
+          ),
+          " "
+        )
+    } else if (gender_age_disagg) {
+      df_data <- df_data %>%
+        separate(disaggregation,
+          into = c(
+            "Sex",
+            "Age"
+          ),
+          " "
+        )
+    } else if (pop_group_disagg) {
+      df_data <- df_data %>%
+        rename(`Population group` = "disaggregation")
+    } else {
+      df_data <- df_data %>%
+        select(-disaggregation)
+    }
+
+    df_msna <- df_msna %>%
+      filter(adm_pcode %in% df_data$adm_pcode) %>%
+      select(-adm0_pcode) %>%
+      arrange(
+        adm_pcode,
+        uuid,
+        sector
+      )
+
+    df_area_name <- df %>%
+      filter(
+        !is.na(adm_pcode),
+        adm0_pcode %in% df_data$adm0_pcode
+      ) %>%
+      transmute(
+        adm1_pcode = gsub("[A-Z]+", "XY", adm1_pcode),
+        adm2_pcode = gsub("[A-Z]+", "XY", adm2_pcode),
+        adm3_pcode = gsub("[A-Z]+", "XY", adm3_pcode),
+        adm_pcode
+      ) %>%
+      unique()
+
+    if (admin_level == 2) {
+      df_area_name <- df_area_name %>%
+        select(-adm3_pcode)
+    }
+
+    if (admin_level < 2) {
+      df_area_name <- df_area_name %>%
+        select(-adm2_pcode)
+    }
+
+    df_data_output <- left_join(
+      df_area_name,
+      df_data
+    ) %>%
+      select(-adm_pcode, -adm0_pcode) %>%
+      filter(!is.na(affected_population)) %>%
+      rename(`Affected population` = affected_population) %>%
+      arrange(
+        adm1_pcode
+      ) %>%
+      mutate(
+        ID = row_number(),
+        .before = "adm1_pcode"
+      )
+
+    names(df_data_output) <- gsub(
+      "adm",
+      "Admin",
+      names(df_data_output)
+    )
+
+    names(df_data_output) <- gsub(
+      "_pcode",
+      " Pcode",
+      names(df_data_output)
+    )
+
+    df_msna_output <- full_join(
+      df_area_name,
+      df_msna
+    ) %>%
+      select(-adm_pcode) %>%
+      filter(!is.na(uuid)) %>%
+      pivot_wider(
+        names_from = sector,
+        values_from = severity
+      ) %>%
+      rename(hh_id = "uuid") %>%
+      mutate(
+        across(
+          .cols = -c(matches("adm"), hh_id, weight),
+          ~ case_when(
+            . >= 3 ~ "in need",
+            . < 3 ~ "not in need",
+            is.na(.) ~ "not applicable"
+          )
+        )
+      )
+
+    names(df_msna_output) <- gsub(
+      "adm",
+      "Admin",
+      names(df_msna_output)
+    )
+
+    names(df_msna_output) <- gsub(
+      "_pcode",
+      " Pcode",
+      names(df_msna_output)
+    )
+
+    write_csv(
```

Comment:
Don't put the CSV overwriting within the function. This prevents us from easily producing multiple versions using different disaggregations.